pkg/eventmonitor/config/config.go

```diff
@@ -0,0 +1,282 @@
+
+	// StatsTagsCardinality determines the cardinality level of the tags added to the exported metrics
+	StatsTagsCardinality string
+
+	// StatsdAddr defines the statsd address
+	StatsdAddr string
+
+	// AgentMonitoringEvents determines if the monitoring events of the agent should be sent to Datadog
+	AgentMonitoringEvents bool
+
+	// CustomSensitiveWords defines words to add to the scrubber
+	CustomSensitiveWords []string
+
+	// ERPCDentryResolutionEnabled determines if the ERPC dentry resolution is enabled
+	ERPCDentryResolutionEnabled bool
+
+	// MapDentryResolutionEnabled determines if the map resolution is enabled
+	MapDentryResolutionEnabled bool
+
+	// DentryCacheSize is the size of the user space dentry cache
+	DentryCacheSize int
+
+	// RemoteTaggerEnabled defines whether the remote tagger is enabled
+	RemoteTaggerEnabled bool
+
+	// NOTE(safchain) need to revisit this one as it can impact multiple event consumers
+	// EnvsWithValue lists environnement variables that will be fully exported
+	EnvsWithValue []string
+
+	// RuntimeMonitor defines if the Go runtime and system monitor should be enabled
+	RuntimeMonitor bool
+
+	// EventStreamUseRingBuffer specifies whether to use eBPF ring buffers when available
+	EventStreamUseRingBuffer bool
+
+	// EventStreamBufferSize specifies the buffer size of the eBPF map used for events
+	EventStreamBufferSize int
+
+	// RuntimeCompilationEnabled defines if the runtime-compilation is enabled
+	RuntimeCompilationEnabled bool
+
+	// EnableRuntimeCompiledConstants defines if the runtime compilation based constant fetcher is enabled
+	RuntimeCompiledConstantsEnabled bool
+
+	// RuntimeCompiledConstantsIsSet is set if the runtime compiled constants option is user-set
+	RuntimeCompiledConstantsIsSet bool
+
+	// NetworkLazyInterfacePrefixes is the list of interfaces prefix that aren't explicitly deleted by the container
+	// runtime, and that are lazily deleted by the kernel when a network namespace is cleaned up. This list helps the
+	// agent detect when a network namespace should be purged from all caches.
+	NetworkLazyInterfacePrefixes []string
+
+	// NetworkClassifierPriority defines the priority at which CWS should insert its TC classifiers.
+	NetworkClassifierPriority uint16
+
+	// NetworkClassifierHandle defines the handle at which CWS should insert its TC classifiers.
+	NetworkClassifierHandle uint16
+
+	// ProcessConsumerEnabled defines if the process-agent wants to receive kernel events
+	ProcessConsumerEnabled bool
+
+	// NetworkConsumerEnabled defines if the network tracer system-probe module wants to receive kernel events
+	NetworkConsumerEnabled bool
+
+	// NetworkEnabled defines if the network probes should be activated
+	NetworkEnabled bool
+
+	// StatsPollingInterval determines how often metrics should be polled
+	StatsPollingInterval time.Duration
+}
+
+// NewConfig creates a config for the event monitoring module
+func NewConfig(spConfig *config.Config) (*Config, error) {
```

Comment:
With this drastic change in the way we define the config, have we ensured that all the proper helm charts (internal and public) have been updated and are properly pinned where they need to be ?

Similarly, how do we plan to notify customers about this change ? I don't think we should expect them to read the release notes or read the agent code, so how do we ensure a smooth transition and backward compatibility to prevent this refactor from breaking CWS customers setup ? (I'm specifically worried about the Helm chart version X agent version compatibility matrix here).

(I'm saying this because some parameters here have a pretty significant impact for customers that use them: I'm namely thinking about the eRPC dentry resolution one, the cache sizes, the network classifiers parameters, etc)